caas/kubernetes/provider/k8s.go

```diff
@@ -1794,6 +1794,14 @@ func (k *kubernetesClient) deleteVolumeClaims(appName string, p *core.Pod) ([]st
 }
 
 func caasServiceToK8s(in caas.ServiceType) (core.ServiceType, error) {
+	// TODO(juju4): remove k8s compatibility fallback
+	// There was no validation at deploy so the raw k8s types may have been used.
+	switch string(in) {
+	case string(core.ServiceTypeClusterIP),
+		string(core.ServiceTypeLoadBalancer),
+		string(core.ServiceTypeExternalName):
+		return core.ServiceType(in), nil
+	}
 	serviceType := defaultServiceType
 	if in != "" {
 		switch in {
```

Comment:
Could we grab validation a new function, then it could be used for apiserver/facades/client/application/application.go

Reply:
added upgrade step so we can now reuse this method without the fallback